frontend/apps/desktop/src/app-onboarding-store.ts

```diff
@@ -0,0 +1,80 @@
+import {DESKTOP_APPDATA} from '@shm/shared/constants'
+import {app, ipcMain} from 'electron'
+import Store from 'electron-store'
+import path from 'path'
+import type {
+  OnboardingFormData,
+  OnboardingState,
+  OnboardingStep,
+} from './app-onboarding'
+
+const ONBOARDING_INITIAL_STATE: OnboardingState = {
+  hasCompletedOnboarding: false,
+  hasSkippedOnboarding: false,
+  currentStep: 'welcome',
+  formData: {
+    name: '',
+  },
+}
+const store = new Store<OnboardingState>({
+  name: 'onboarding',
+  cwd: path.join(app?.getPath('userData') || process.cwd(), DESKTOP_APPDATA),
+  defaults: ONBOARDING_INITIAL_STATE,
+})
+
+const getInitialState = (): OnboardingState => ONBOARDING_INITIAL_STATE
+
+export function setupOnboardingHandlers() {
+  ipcMain.on('get-onboarding-state', (event) => {
```

Comment:
wouldn't it be easier to use TRPC like we do for everything else?

Like recents, favorites, settings, the "host" store stuff in my other branch, gatewaySettings, recentSigners, drafts, commentDrafts.

I don't understand why we are breaking convention here by using ipc directly instead of TRPC+invalidation. It seems very inconvenient and inconsistent in this codebase.

Reply:
like I said above, I want the minimal things as possible in the middle end. and when I was using TRPC I was getting flashes on the first load of the app